examples/plot_credit_loan_decisions.py

```diff
@@ -0,0 +1,1293 @@
+# will introduce a semi-synthetic feature into an existing publicly available
+# dataset to replicate the outcome disparity found in the original dataset.
+
+# %%
+# Credit decisions dataset
+# ========================
+
+# %%
+# As mentioned, we will not be able to use the original loans dataset, and
+# instead will be working with a publicly available dataset of credit card
+# defaults in Taiwan collected in 2005. This dataset represents binary credit
+# card default outcomes for 30,000 applicants with information pertaining to an
+# applicant's payment history and bill statements over a six-month period from
+# April 2005 to September 2005, as well as demographic information, such as
+# *sex*, *age*, *marital status*, and *education level* of the applicant. A full
+# summary of features is provided below:
+# 
+# .. list-table::
+#   :header-rows: 1
+#   :stub-columns: 1
+# 
+#   *  - features
+#      - description
+#   *  - sex, education, marriage, age
+#      - demographic features
+#   *  - pay_0, pay_2, pay_3, pay_4, pay_5, pay_6 
+#      - repayment status (ordinal)
+#   *  - bill_amt1, bill_amt2, bill_amt3, bill_amt4, bill_amt5, bill_amt_6
+#      - bill statement amount (Taiwan dollars)
+#   *  - pay_amt1, pay_amt2, pay_amt3, pay_amt4, pay_amt5, pay_amt6
+#      - previous statement amount (Taiwan dollars)
+#   *  - default payment next month
+#      - default information (1 = YES, 0 = NO)
+# 
+# Let's pretend we are a data scientist at a financial institution who is
+# tasked with developing a classification model to predict whether an applicant
+# will default on a personal loan. A positive prediction by the model means the
+# applicant would default on the credit loan. *Defaulting on a loan* means the
+# client fails to make payments within a 30-day window, and the lender can take
+# legal actions against the client.
+#
+# Although we do not have a dataset of loan default history, we do have this
+# data set of credit card payment history. We assume customers who make monthly
+# credit card payments on time are more *creditworthy*, and thus less likely to
+# default on a personal credit loan.
+#
+# **Decision point: task definition**
+#
+# - **Defaulting on a credit card payment** can be viewed as a proxy for the
+#   fact that an applicant might not be a good candidate for a personal loan.
+# - Because most customers did not default on their credit card payment, we will
+#   need to take this class imbalance into account during our modeling process.
+#
+# As the data is read in-memory, we will change the column :code:`PAY_0` to
+# :code:`PAY_1` to make the naming more consistent with the naming of the other
+# columns. In addition, the target variable :code:`default payment next month`
+# is changed to :code:`default` to reduce verbosity.
+
+# %%
+data_url = "http://archive.ics.uci.edu/ml/machine-learning-databases/00350/default%20of%20credit%20card%20clients.xls"
```

Comment:
Did we want to use the newly added `fetch_credit_card` instead of this?

Reply:
The problem with `fetch_credit_card` is that these are the column names:
` x1   x2   x3   x4    x5   x6   x7   x8   x9  x10  x11       x12       x13       x14      x15      x16      x17      x18      x19      x20     x21      x22     x23`
Should we fix this in the definition of `fetch_credit_card`?

Reply:
uff. this makes me worried about `fetch_credit_card`. i am fine merging as is, but i'll open an issue to fix the feature names. we have the same problem with `fetch_bank_marketing`.

Reply:
See #1165